curator-framework/src/main/java/org/apache/curator/framework/CuratorFrameworkFactory.java

```diff
@@ -474,6 +477,28 @@ public Builder schemaSet(SchemaSet schemaSet)
             return this;
         }
 
+        /**
+         * Curator (and user) recipes will use this executor to call notifyAll
+         * and other blocking calls that might normally block ZooKeeper's event thread.
+         * By default, an executor is allocated internally using the provided (or default)
+         * {@link #threadFactory(java.util.concurrent.ThreadFactory)}. Use this method
+         * to set a custom executor.
+         *
+         * @param runSafeService executor to use for calls to notifyAll from Watcher callbacks etc
+         * @return this
+         * @since 4.1.0
+         */
+        public Builder runSafeService(Executor runSafeService)
+        {
+            this.runSafeService = runSafeService;
+            return null;
```

Comment:
This should return this rather than null

Reply:
damn - good catch